### PR TITLE
feat: add story telling workshop

### LINKED ISSUE #621: chore: add storytelling workshop to main repo
Here is the prototype

https://github.com/freeCodeCamp/CurriculumExpansion/pull/449/files

We need to add this workshop with steps and tests to the main repo.

### PR DESCRIPTION
**Pull request by zairahira**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes https://github.com/freeCodeCamp/CurriculumExpansion/issues/621

### CODE REVIEW

**Comment by moT01** [comment]
A PR has been merged that renamed the certification from "Front End" to "Full Stack". This PR may need an update and a few changes made: [see comment](https://github.com/freeCodeCamp/freeCodeCamp/pull/56694#issuecomment-2444473378)

**Comment by moT01** [comment]
A PR has been merged that moves the block order from the `meta.json` file to the `full-stack.json` file.](https://github.com/freeCodeCamp/freeCodeCamp/pull/56795) This PR will need an update, and the order will need to be removed from the `meta.json` file and moved to the new file.

**Comment by jdwilkin4** [comment]
Here is how you can test the workshops locally

https://contribute.freecodecamp.org/how-to-work-on-coding-challenges/#testing-challenges

**Review by Dario-DC — COMMENTED** [review]
It's looking good, I left a few comments. I was having issues with the last step because of a missing space in the seed code. I should have reported all the occurrences.

File: curriculum/challenges/english/25-front-end-development/workshop-story-teller-app/671fafbae959c048d82bac8d.md [hunk 1]
```diff
@@ -0,0 +1,379 @@
+---
+id: 671fafbae959c048d82bac8d
+title: Step 16
+challengeType: 0
+dashedName: step-16
+---
+
+# --description--
+
+Add a `click` event listener to the `scaryStoryBtn`, `funnyStoryBtn`, and `adventureStoryBtn` buttons.
```
[the remark this answers is not in the document]
Comment (zairahira) [reply curriculum/challenges/english/25-front-end-development/workshop-story-teller-app/671fafbae959c048d82bac8d.md]: done

File: curriculum/challenges/english/25-front-end-development/workshop-story-teller-app/671fa22e99042b1e91fe9d2e.md [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+assert.lengthOf(document.querySelectorAll('.btn-container > button') ,3);
+```
+
+All the three buttons should have a class of `btn`.
+
+```js
+assert.equal(document.querySelectorAll('.btn-container > .btn').length, 3);
+```
+
+You should have a button with the `id` of `scary-btn` and a text of `"Scary Story"`.
```
Comment (Dario-DC) [inline curriculum/challenges/english/25-front-end-development/workshop-story-teller-app/671fa22e99042b1e91fe9d2e.md:26]:
```suggestion
You should have a button with the `id` of `scary-btn` and a text of `Scary Story`.
```

File: curriculum/challenges/english/25-front-end-development/workshop-story-teller-app/671fa22e99042b1e91fe9d2e.md [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+
+```js
+assert.exists(document.querySelector('#scary-btn'));
+assert.equal(document.querySelector('#scary-btn').textContent, 'Scary Story');
+```
+
+You should have a button with the `id` of `funny-btn` and a text of `"Funny Story"`.
```
Comment (Dario-DC) [inline curriculum/challenges/english/25-front-end-development/workshop-story-teller-app/671fa22e99042b1e91fe9d2e.md:33]:
```suggestion
You should have a button with the `id` of `funny-btn` and a text of `Funny Story`.
```

File: curriculum/challenges/english/25-front-end-development/workshop-story-teller-app/671fa22e99042b1e91fe9d2e.md [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+
+```js
+assert.exists(document.querySelector('#funny-btn'));
+assert.equal(document.querySelector('#funny-btn').textContent, 'Funny Story');
+```
+
+You should have a button with the `id` of `adventure-btn` and a text of `"Adventure Story"`.
```
Comment (Dario-DC) [inline curriculum/challenges/english/25-front-end-development/workshop-story-teller-app/671fa22e99042b1e91fe9d2e.md:40]:
```suggestion
You should have a button with the `id` of `adventure-btn` and a text of `Adventure Story`.
```

File: curriculum/challenges/english/25-front-end-development/workshop-story-teller-app/671fa82dc39e4c320352f371.md [hunk 5]
```diff
@@ -0,0 +1,212 @@
+---
+id: 671fa82dc39e4c320352f371
+title: Step 9
+challengeType: 0
+dashedName: step-9
+---
+
+# --description--
+
+To hold the different types of stories, create an object called `storyObj`. The object should have three properties: `scary`, `funny`, and `adventure`. Each property should have a `story` and a `borderColor` property.
```
Comment (Dario-DC) [inline curriculum/challenges/english/25-front-end-development/workshop-story-teller-app/671fa82dc39e4c320352f371.md:10]:
```suggestion
To hold the different types of stories, create an object called `storyObj`. The object should have three properties: `scary`, `funny`, and `adventure`. Each property value should be an object with a `story` and a `borderColor` property.
```

File: curriculum/challenges/english/25-front-end-development/workshop-story-teller-app/671fae2d8e402a405a930e44.md [hunk 6]
```diff
@@ -0,0 +1,172 @@
+---
+id: 671fae2d8e402a405a930e44
+title: Step 11
+challengeType: 0
+dashedName: step-11
+---
+
+# --description--
+
+Add the `displayStory` function to the `click` event listener of the `scaryStoryBtn`, and see the output in the console.
```
Comment (Dario-DC) [inline curriculum/challenges/english/25-front-end-development/workshop-story-teller-app/671fae2d8e402a405a930e44.md:10]:
```suggestion
Add an event listener for the `click` event to `scaryStoryBtn` and pass it the `displayStory` function. Then, click your button and see the output in the console.
```

File: curriculum/challenges/english/25-front-end-development/workshop-story-teller-app/671fae2d8e402a405a930e44.md [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+
+# --hints--
+
+You should add the `displayStory` function to the `click` event listener of the `scaryStoryBtn` and see the output `"You clicked the button"` in the console when you click the `scaryStoryBtn`.
```
Comment (Dario-DC) [inline curriculum/challenges/english/25-front-end-development/workshop-story-teller-app/671fae2d8e402a405a930e44.md:14]:
```suggestion
You should add an event listener for the `click` event to `scaryStoryBtn` passing it the `displayStory` function.
```

File: curriculum/challenges/english/25-front-end-development/workshop-story-teller-app/671faf4ac1e19245df4db7f5.md [hunk 8]
```diff
@@ -0,0 +1,196 @@
...
+title: Step 14
+challengeType: 0
+dashedName: step-14
+---
+
+# --description--
+
+Inside the `displayStory` function, first check if the `storyObj` object has a property that matches the `genre` parameter.
+
+If the `storyObj` object has a property that matches the `genre` parameter, set the `#result` paragraph's text content to the story of the selected genre.
```
Comment (Dario-DC) [inline curriculum/challenges/english/25-front-end-development/workshop-story-teller-app/671faf4ac1e19245df4db7f5.md:12]:
```suggestion
Inside the `displayStory` function, check if the `storyObj` object has a property that matches the `genre` parameter and set the `#result` paragraph's text content to the story of the selected genre.
```

File: curriculum/challenges/english/25-front-end-development/workshop-story-teller-app/671faf4ac1e19245df4db7f5.md [hunk 9]
```diff
@@ -0,0 +1,196 @@
...
+
+Also, set the `storyContainer`'s border color to the `borderColor` property of the selected genre.
+
+Test this by calling the `displayStory` function with the `scary` genre. The border colour should be `#ee4b2b` and the story should be `"In the dark woods, a group of friends stumbled upon an old, abandoned cabin. They enter the cabin and awaken something malevolent that had been dormant for centuries."`.
```
Comment (Dario-DC) [inline curriculum/challenges/english/25-front-end-development/workshop-story-teller-app/671faf4ac1e19245df4db7f5.md:16]:
```suggestion
Test this by calling the `displayStory` function with the `scary` genre. The border color should be `#ee4b2b` and the story should be `"In the dark woods, a group of friends stumbled upon an old, abandoned cabin. They enter the cabin and awaken something malevolent that had been dormant for centuries."`.
```

**Review by Dario-DC — COMMENTED** [review]
Left a few comments. After those are resolved it's ready to go in :+1:

**Review by Dario-DC — APPROVED** [review]

File: curriculum/challenges/english/25-front-end-development/workshop-storytelling-app/671fa537114e412950b62089.md [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+dashedName: step-6
+---
+
+# --description--
+
+In your JavaScript file, select the `.story-container` `div` and store it in a variable called `storyContainer`.
+
+# --hints--
+
+Your `storyContainer` variable should have the value of `document.querySelector(".story-container")`.
```
Comment (jdwilkin4) [inline curriculum/challenges/english/25-front-end-development/workshop-storytelling-app/671fa537114e412950b62089.md:14]: We shouldn't give them the exact code here.

We should rephrase this to hint that they will need to access the DOM using the querySelector method.
Also, there should be a test before this to check that they created the variable first.

File: curriculum/challenges/english/25-front-end-development/workshop-storytelling-app/671fa769540d152e53327f4b.md [hunk 11]
```diff
@@ -0,0 +1,147 @@
...
+
+```js
+assert.deepEqual(funnyStoryBtn, document.getElementById("funny-btn"));
+```
+
+Your `adventureStoryBtn` should have the value of `document.getElementById("adventure-btn")`.
+
+```js
+assert.deepEqual(adventureStoryBtn, document.getElementById("adventure-btn"));
+```
```
Comment (jdwilkin4) [inline curriculum/challenges/english/25-front-end-development/workshop-storytelling-app/671fa769540d152e53327f4b.md:30]: The directions don't mention that you have to use `getElementById`. but `querySelector` is also a valid answer.
The hints need to be updated to account for that.

Also, the tests should be more granular. 
We should have tests to check that each of the variables exists, then check that the correct value has been assigned to it.

File: curriculum/challenges/english/25-front-end-development/workshop-storytelling-app/671fa7ed8364b130b5c55d2e.md [hunk 12]
```diff
@@ -0,0 +1,142 @@
...
+
+Select the `#result` paragraph and store it in a variable called `resultParagraph`. You will use this variable to display the selected story.
+
+# --hints--
+
+Your `resultParagraph` variable should have the value of `document.getElementById("result")`.
+
+```js
+assert.deepEqual(resultParagraph, document.getElementById("result"));
+```
```
Comment (jdwilkin4) [inline curriculum/challenges/english/25-front-end-development/workshop-storytelling-app/671fa7ed8364b130b5c55d2e.md:18]: Same here.
querySelector is also a valid option.
So the hint needs to be updated to account for that

**Review by jdwilkin4 — CHANGES_REQUESTED** [review]
Left a couple of notes 👍

File: curriculum/challenges/english/25-front-end-development/workshop-storytelling-app/671fa537114e412950b62089.md (hunk 10 above)
Comment (zairahira) [reply curriculum/challenges/english/25-front-end-development/workshop-storytelling-app/671fa537114e412950b62089.md]: Done!

File: curriculum/challenges/english/25-front-end-development/workshop-storytelling-app/671fa7ed8364b130b5c55d2e.md (hunk 12 above)
Comment (zairahira) [reply curriculum/challenges/english/25-front-end-development/workshop-storytelling-app/671fa7ed8364b130b5c55d2e.md:18]: Done!

File: curriculum/challenges/english/25-front-end-development/workshop-storytelling-app/671fa769540d152e53327f4b.md (hunk 11 above)
Comment (zairahira) [reply curriculum/challenges/english/25-front-end-development/workshop-storytelling-app/671fa769540d152e53327f4b.md]: Done!

File: client/i18n/locales/english/intro.json [hunk 13]
```diff
@@ -2808,6 +2808,12 @@
         ]
       },
       "gibb": { "title": "188", "intro": [] },
+      "workshop-storytelling-app": {
+        "title": "Build a Storytelling App",
+        "intro": [
+          "In this workshop, you will Build a Storytelling App that will allow you to list different stories based on genre."
```
Comment (jdwilkin4) [inline client/i18n/locales/english/intro.json:2814]:
```suggestion
          "In this workshop, you will build a storytelling app that will allow you to list different stories based on genre."
```

**Review by jdwilkin4 — APPROVED** [review]
Looks good 👍
